### PR TITLE
add description about pod nanny

### LINKED ISSUE #481: Document resource limit configuration
/kind feature

**What happened**:
I notice that kube-state-metrics has its on magic in terms of resource governing thru the "pod-nanny" sidecar. It's unclear to me how to tune resource settings for kube-state-metrics, since I gather it's not done using the regular `resources` attribute on the deployment.

**What you expected to happen**:
I'd like to find clear documentation on how to set up memory/cpu limits using the supplied kubernetes manifests.

### PR DESCRIPTION
**Pull request by andyxning**
**What this PR does / why we need it**:
This PR will add description about the `addon-resizer` component.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #481

### CODE REVIEW

**Comment by andyxning** [comment]
Ping @trondhindenes Please take a look at this whether this description can alleviate your doubt.

File: README.md [hunk 1]
```diff
@@ -113,7 +113,13 @@ kube-state-metrics exposes its own metrics under `--telemetry-host` and `--telem
 
 ### Resource recommendation
 
-Resource usage changes with the size of the cluster. As a general rule, you should allocate
+Resource usage for kube-state-metrics changes with the size, i.e., the node number, of the cluster.
```
Comment (brancz) [inline README.md]: I think it's actually a problem that it doesn't exactly scale with the number of nodes, but rather with the total number of objects, which can vastly vary even at the same node count. I think we should mention that here.

**Comment by andyxning** [comment]
@brancz PTAL.

**Comment by brancz** [comment]
/lgtm

thanks!

**Comment by trondhindenes** [comment]
Sorry for not seeing this before - looks good, thanks so much for adding!

**Pull request merged**